### PR TITLE
add test generators and infrastructure

### PR DESCRIPTION
**Pull request by acruikshank**
closes https://www.pivotaltracker.com/story/show/177633808

### Motivation

We're starting to recognize a common need for random test helpers across Liberty repos. This PR creates a package to implement several of those functions so they may be used across repos without code duplication. This code is mostly stolen from [sdk-ts](/LibertyDSNP/sdk-ts) and [example-client](/LibertyDSNP/example-client).

### Potential Issues

1. A couple of the the test helpers want to return a `HexString`. This is a type alias of `string`. I've redefined that alias here. I'm not really sure how that's going to play in other repos. If it conflicts, it's probably best to just have these functions return `string`.
2. The test functions that return a `HexString` prefix the string with '0x'. This looks more like a string representation of a hex integer literal than something that is natively a hex string. It seems wrong, but I've left it for compatibility.
3. The test functions are truly random, so any test that uses them are non-deterministic or semi-non-deterministic if the tests are seeded early. This could cause flaky test behavior at some point.

### What's in this PR

1. Add generators.ts with several generators from [example-client](/LibertyDSNP/example-client).
2. Add tests for the generators.
3. Add `HexString` type (see potential issue above)
4. Add README, LICENSE, dependencies, linting, docs, etc to make a typescript proper library.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Library Development Notes
+- Update the docs folder as needed by running `npm run doc:markdown`
+
+### Local Development Library Use
+
+1. test-generators: `npm install`
+2. test-generators: `npm run build`
+3. test-generators: `npm link`
+4. In other project: `npm link @LibertyDSNP/test-generators`
```
Comment (wilwade) [inline CONTRIBUTING.md]:
```suggestion
4. In other project: `npm link @unfinishedlabs/test-generators`
```

File: src/types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export type HexString = string;
```
Comment (wilwade) [inline src/types/index.d.ts:1]: > A couple of the the test helpers want to return a HexString. This is a type alias of string. I've redefined that alias here. I'm not really sure how that's going to play in other repos. If it conflicts, it's probably best to just have these functions return string.

TypeScript will handle this as a string. A type alias is just for better naming. When type checking there is no difference from the type `string`.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Test Generators
```
Comment (wilwade) [inline README.md:1]: Could you also drop in a `.tool-versions` file for asdf and a `.node-version` for nvm users?

https://github.com/LibertyDSNP/sdk-ts/blob/main/.tool-versions

File: package.json [hunk 4]
```diff
@@ -0,0 +1,88 @@
+{
+  "name": "@unfinishedlabs/test-generators",
+  "version": "0.0.0",
+  "description": "Utilities to generate test data",
+  "type": "commonjs",
+  "main": "dist/cjs/index.js",
+  "types": "dist/types/index.d.ts",
+  "module": "dist/esm/index.js",
```
Comment (wilwade) [inline package.json:8]: These currently point to `index`, but we are only generating `generators`. We should keep this, but change `generators.ts` to `index.ts`.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+Problem
```
Comment (wilwade) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Do we want to pull over the issues templates and github actions as well?

**Review by wilwade — CHANGES_REQUESTED** [review]
A few small comments.

**Comment by enddynayn** [comment]
👍

**Review by enddynayn — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (acruikshank) [reply README.md:1]: done.

File: package.json (hunk 4 above)
Comment (acruikshank) [reply package.json:8]: done.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (acruikshank) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: done.

**Review by wilwade — APPROVED** [review]
👍

**Pull request merged**